src/CLIMA-atmos/Parameterizations/SurfaceFluxes/src/SurfaceFluxes.jl

```diff
@@ -0,0 +1,145 @@
+"""
+    SurfaceFluxes
+
+Surface flux functions, e.g., for buoyancy flux,
+friction velocity, and exchange coefficients.
+
+"""
+module SurfaceFluxes
+
+using Utilities.RootSolvers
+using Utilities.MoistThermodynamics
+using PlanetParameters
+
+function compute_buoyancy_flux(
+  shf::R,
+  lhf::R,
+  T_b::R,
+  qt_b::R,
+  ql_b::R,
+  qi_b::R,
+  alpha0_0::R
+  )::R where R
+  cp_ = cp_m(qt_b, ql_b, qi_b)
+  lv = latent_heat_vapor(T_b)
+  temp1 = (PlanetParameters.eps_vi-1.0)
+  temp2 = (shf + temp1 * cp_ * T_b * lhf /lv)
+  return (PlanetParameters.grav * alpha0_0 / cp_ / T_b * temp2)
+end
+
+function ψ_m_unstable(ζ::R, ζ_0::R) where R
+  x(ζ) = (1.0 - PlanetParameters.gamma_m * ζ)^0.25
+  temp(ζ, ζ_0) = log((1.0 + ζ)/(1.0 + ζ_0))
+  ψ_m = (2.0 * temp(x(ζ), x(ζ_0)) + temp(x(ζ)^2, x(ζ_0)^2) - 2.0 * atan(x(ζ)) + 2.0 * atan(x(ζ_0)))
+  return ψ_m
+end
+
+function ψ_h_unstable(ζ::R, ζ_0::R) where R
+  y(ζ) = sqrt(1.0 - PlanetParameters.gamma_h * ζ )
+  ψ_h = 2.0 * log((1.0 + y(ζ))/(1.0 + y(ζ_0)))
+  return ψ_h
+end
+ψ_m_stable(ζ::R, ζ_0::R) where R = -PlanetParameters.beta_m * (ζ - ζ_0)
+ψ_h_stable(ζ::R, ζ_0::R) where R = -PlanetParameters.beta_h * (ζ - ζ_0)
+
+"""
+Computes roots of equation (eq. 10 in Ref. 1)
+
+  windspeed = u_* ( ln(z/z_0) - ψ_m(z/Λ, z_0/Λ) ) /κ        eq. 10 in Ref. 1
+  Λ = -u_*^3/(buoyancy_flux κ)                              eq. 22 in Ref. 2
+  ψ_m(ζ, ζ_0) = [ψ_m_stable, ψ_m_unstable]                  eq. 12, 14 in Ref 1
+
+Ref 1: Byun, Daewon W. "On the analytical solutions of flux-profile
+       relationships for the atmospheric surface layer." Journal of
+       Applied Meteorology 29.7 (1990): 652-657.
+       https://journals.ametsoc.org/doi/pdf/10.1175/1520-0450%281990%29029%3C0652%3AOTASOF%3E2.0.CO%3B2
+
+Ref 2: Wyngaard, John C. "Modeling the planetary boundary layer-Extension
+       to the stable case." Boundary-Layer Meteorology 9.4 (1975): 441-460.
+       https://sci-hub.tw/https://link.springer.com/article/10.1007/BF00223393
+"""
```

Comment:
Please use the finite-volume versions of the relevant expressions in Nishizawa and Kitamura (2018). You might want to think about how to be able to use both, the finite-difference and finite-volume versions.